### DISCUSSION TITLE
How to use hasura derive actions?

### DISCUSSION THREAD

**Discussion by rbansal9148** [hasura/graphql-engine#7840] [Question]
Hello Everyone,

I have a insert query, which should be validated before execution.
1. So I wrote the query graphiQL and clicked `Derive Action`.
2. Now it redirects me to a page of action creation with everything prefilled and I save it. At this point if i run the query, it directly inserts to the table. Have configured the perm for the role I am using.
3. Now I have to stop the direct table insertions, so i enabled the backend only flag on the insert query and wrote the handler in my code.

After all this if i run the query it throws an error `"field \"insert_${tableName}_one\" not found in type: 'mutation_root'"`.

Can you guys help me to figure out what I am doing wrong?

hasura version: 2.1.0-beta.2/2.0.9 (tried on both)

**Comment by BenoitRanque** [comment]
Derived actions do not replace the element they are derived from.
When you defined your action you also gave it a name with which to identify it within the root mutation object.
